### PR TITLE
[fix](nereids)subquery unnesting get wrong result if correlated conjuncts is not slot_a = slot_b

### PR DESCRIPTION
**Pull request by starocean999**
sql:
`select * from t1 where t1.c1 < (select max(t2.c1) from t2 where t1.c2 = abs(t2.c2));`
before this pr, we would use the equivalent sql as bellow, which is wrong
`select t1.* from t1 semi join ( select t2.c2 as c2, max(t2.c1) as c1 from t2 group by t2.c ) tx on t1.c1 < tx.c1 and t1.c2 = tx.c2;`
the pr fix the problem and get the correct one as:
`select t1.* from t1 semi join ( select abs(t2.c2) as c2, max(t2.c1) as c1 from t2 group by abs(t2.c) ) tx on t1.c1 < tx.c1 and t1.c2 = tx.c2;`

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Rewriter.java [hunk 1]
```diff
@@ -204,7 +204,8 @@ public class Rewriter extends AbstractBatchJobExecutor {
                                  *  TODO: group these rules to make sure the result plan is what we expected.
                                  */
                                 new CorrelateApplyToUnCorrelateApply(),
-                                new ApplyToJoin()
+                                new ApplyToJoin(),
+                                new NormalizeAggregate()
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Rewriter.java:208]: why need this normalize aggregate?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/UnCorrelatedApplyAggregateFilter.java [hunk 2]
```diff
@@ -97,10 +100,19 @@ public List<Rule> buildRules() {
...
+        Map<Expression, Slot> unCorrelatedExprToSlot = Maps.newHashMap();
+        newAggOutput.addAll(newGroupby.stream().map(expression -> {
+            if (expression instanceof Slot) {
+                return (NamedExpression) expression;
+            } else {
+                Alias alias = new Alias(expression);
+                unCorrelatedExprToSlot.put(expression, alias.toSlot());
+                return alias;
+            }
+        }).collect(ImmutableList.toImmutableList()));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/UnCorrelatedApplyAggregateFilter.java:114]: do not use stream api

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Rewriter.java (hunk 1 above)
Comment (starocean999) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Rewriter.java:208]: UnCorrelatedApplyAggregateFilter rule will create new aggregate outputs, and the later rule CheckPrivileges which inherent from ColumnPruning depends on NormalizeAggregate.  There may be some better way to make it work, but correctly, use NormalizeAggregate for simplicity.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/UnCorrelatedApplyAggregateFilter.java (hunk 2 above)
Comment (starocean999) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/UnCorrelatedApplyAggregateFilter.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Rewriter.java (hunk 1 above)
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Rewriter.java:208]: why ColumnPruning depends on NormalizeAggregate?

**Comment by starocean999** [comment]
run External

**Comment by starocean999** [comment]
run external

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**